experiment/simulation/js/main.js

```diff
@@ -1 +1,376 @@
+
+	slider_vibe.oninput = function() {
+		output_vibe.innerHTML = this.value;
+		vibe = Number(document.getElementById("vibration").value);
+		restart();
+	};
+
+	const slider_stiff = document.getElementById("stiff");
+	const output_stiff = document.getElementById("demo_stiff");
+	output_stiff.innerHTML = slider_stiff.value; // Display the default slider value
+
+	slider_stiff.oninput = function() {
+		output_stiff.innerHTML = this.value;
+		stiffness = Number(document.getElementById("stiff").value);
+		restart();
+	};
+
+	const slider_etta = document.getElementById("etta");
+	const output_etta = document.getElementById("demo_etta");
+	output_etta.innerHTML = slider_etta.value; // Display the default slider value
+
+	slider_etta.oninput = function() {
+		output_etta.innerHTML = this.value;
+		etta = Number(document.getElementById("etta").value);
+		etta /= 100;
+		restart();
+	};
+
+	function drawGround(ctx, ground)
+	{
+		ctx.save();
+		ctx.fillStyle = "pink";
+		ctx.beginPath();
+		ctx.moveTo(ground[0][0], ground[0][1]);
+
+		for(let i = 0; i < ground.length; ++i)
+		{
+			const next = (i + 1) % ground.length;
+			ctx.lineTo(ground[next][0], ground[next][1]);
+		}
+
+		ctx.closePath();
+		ctx.fill();
+		ctx.stroke();
+		ctx.restore();
+	}
+
+	const height = 100;
+	const startx = 596;
+	const endx = 600;
+	const margin = 150;
+	const starty = 220;
+
+	// Input Parameters 
+	let vibe = 100;	// intial disaplcement
+	let dev = 1000; // intial velocity
+	let mass = 136000; 
+	let stiff = 30 * 1000000;
+	let isdamp = 0;
+	let etta = 0.05;
+
+	let radius = mass / 10000;
+
+	// Derived Parameters
+	let rod, ground, curr_t;
+	init();
+	let omegan = math.sqrt(stiff / mass);
+	let omega0 = omegan * Math.sqrt(1 - (etta * etta));
+	let curr_displacement = vibe * (Math.cos(omegan * curr_t)) + (dev * (Math.sin(omegan * curr_t))) / omegan;
+	let damped_displacement = (Math.exp(-1 * etta * omega0 * curr_t)) * (vibe * (Math.cos(omega0 * curr_t)) + ((dev + etta * omegan * vibe) * Math.sin(omegan * curr_t)) / omegan);
+
+	const canvas = document.getElementById("main");
+	canvas.width = 1200;
+	canvas.height = 600;
+	canvas.style = "border:3px solid";
+	const ctx = canvas.getContext("2d");
+
+	const fill = "#A9A9A9";
+	const border = "black";
+	const lineWidth = 1.5;
+
+	const fps = 15;
+
+	function draw()
+	{
+		ctx.clearRect(0, 0, canvas.width, canvas.height);
+		ctx.fillStyle = fill;
+		ctx.lineWidth = lineWidth;
+		ctx.lineCap = "round";
+		ctx.lineJoin = "round";
+
+		drawGround(ctx, ground);
+		const v = [...rod];
+
+		if(isdamp)
+		{
+			curr_displacement = (Math.exp(-1 * etta * omega0 * curr_t)) * (vibe * (Math.cos(omega0 * curr_t)) + ((dev + etta * omegan * vibe) * Math.sin(omegan * curr_t)) / omega0);
+		}
+		else
+		{
+			curr_displacement = vibe * (Math.cos(omegan * curr_t)) + (dev * (Math.sin(omegan * curr_t))) / omegan;
+		}
+
+		newv = v[0][0] + curr_displacement;
+		newv2 = v[1][0] + curr_displacement;
```

Comment:
What does newv2 signify? Is it a second part of something? Can we give a more descriptive name?